test/integration/02-user-wallet/02-fees-onchain.spec.ts

```diff
@@ -18,15 +29,35 @@ afterAll(async () => {
 
 describe("UserWallet - getOnchainFee", () => {
   it("returns a fee greater than zero for an external address", async () => {
-    const address = await bitcoindOutside.getNewAddress()
-    const fee = await userWallet0.getOnchainFee({ address })
+    const address = (await bitcoindOutside.getNewAddress()) as OnChainAddress
+    const fee = await Wallets.getOnChainFee(userWallet0, defaultAmount, address)
     expect(fee).toBeGreaterThan(0)
+    expect(fee).toBeGreaterThan(userWallet0.withdrawFee)
   })
 
   it("returns zero for an on us address", async () => {
-    const address = await Wallets.createOnChainAddress(userWallet1.user.id)
+    const address = await Wallets.createOnChainAddress(userWallet1.id)
     if (address instanceof Error) throw address
-    const fee = await userWallet0.getOnchainFee({ address })
+    const fee = await Wallets.getOnChainFee(userWallet0, defaultAmount, address)
     expect(fee).toBe(0)
   })
+
+  it("returns error for dust amount", async () => {
```

Comment:
Should return `InvalidSatoshiAmount` (if < 0) and `LessThanDustThreshold` (if < dust) or something like that (so caller can differentiate what the issues i)

Reply:
is a validation of the returned errors (in the method) and cant be in unit test because it depends on db (only should return the error if it is not an intra ledger tx).